src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,244 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createCarQuery = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement saveCarStatement
+                            = connection.prepareStatement(createCarQuery,
+                            Statement.RETURN_GENERATED_KEYS)) {
+            saveCarStatement.setLong(1, car.getManufacturer().getId());
+            saveCarStatement.setString(2, car.getModel());
+            saveCarStatement.executeUpdate();
+            ResultSet resultSet = saveCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create " + car + ". ", throwable);
+        }
+        createCarDriversRelations(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id, cars.model, "
+                + "cars.manufacturer_id, "
+                + "manufacturers.name, "
+                + "manufacturers.country "
+                + "FROM taxi_schema.cars "
+                + "JOIN taxi_schema.manufacturers "
+                + "ON taxi_schema.cars.manufacturer_id = taxi_schema.manufacturers.id "
+                + "WHERE taxi_schema.cars.id = ? AND taxi_schema.cars.is_deleted = 0;";
+        Optional<Car> car = null;
```

Comment:
```suggestion
        Optional<Car> car = Optional.empty();
```

we should return empty Optional, not null